source/Ship.cpp

```diff
@@ -2020,7 +2020,8 @@ bool Ship::FireAntiMissile(const Projectile &projectile, vector<Visual> &visuals
 Point Ship::FireTractorBeam(const Flotsam &flotsam, vector<Visual> &visuals)
 {
 	Point pullVector;
-	if(flotsam.Position().Distance(position) > tractorBeamRange)
+	const double distance = flotsam.Position().Distance(position);
+	if(distance > tractorBeamRange || distance - Radius() <= 0)
```

Comment:
This would cause tractor beams to stop firing pretty early for long ships given that the flotsam is arriving perpendicular to the ship's length. The description says that this is "stopping the tractor beam from trying to pull flotsams that are already in collection range," but the radius of the ship isn't actually the collection range of the flotsam.